### PR TITLE
Bug Fix for Merging "reenhanced:master" instead of "master"

### PR DESCRIPTION
**Pull request by simonzhu24**

### CODE REVIEW

**Comment by simonzhu24** [comment]
@codenamev Apologies for sending another diff, but I found a bug a bit last minute when `deliver` commands trying to deliver on `reenhanced:master` instead of `master`. There was also something else going on with the exclusion of `self` which doesn't make a lot of sense, so I added the `self's` back in. Please accept this asap so the bug can be fixed.

**Comment by codenamev** [comment]
Can you update the tests please?  I have another branch (`vs-cleanup-new-gh-merge`) I've been working on to clean up some other issues I've found as well before we release.

**Comment by simonzhu24** [comment]
The tests are passing as I didn't add any additional methods or complex
logic that would cause them to fail.

There was no added functionality so I don't believe new tests need to be
added?

**Comment by simonzhu24** [comment]
@codenamev do you mind patching in my changes and seeing if the issues that you saw still exist? I believe this patch will solve the clean up and deliver issues that were present.

**Comment by codenamev** [comment]
The tests are there for quality assurance that the methods do what they say.  Since we have restructured so much here, it's possible that the old tests didn't cover the new way of doing things.  We'll want to first add failing tests to catch the issue you found so that we can ensure any new changes going forward don't affect it as well.

Also, we have a [running PR open](https://github.com/reenhanced/gitreflow/pull/172) doing some cleanup before we can release.  It turns out that the squash feature in the API is only in preview and requires special headers to access.  While we can simple add those headers in, the `github_api` gem does not currently support the extra parameters (it filters unwanted options out).  There is a PR we have open to get that resolved and will need it released in the `github_api` gem before we can release this as well.

**Comment by simonzhu24** [comment]
@codenamev I updated the tests.

**Comment by codenamev** [comment]
It looks like the bug in question is using a new syntax for the branch name `owner:base_branch_name`?  Can you please update/add tests for that scenario?

**Comment by simonzhu24** [comment]
@codenamev That is incorrect. The checkout command that were ran after `delivery` were `git checkout owner:base_branch_name` which was incorrect as it should have been `git checkout base_branch_name`. Because some of those commands were left out of the test, I added them back in. Now the code works and the tests pass.

File: lib/git_reflow/git_server/pull_request.rb [hunk 1]
```diff
@@ -169,41 +169,36 @@ def deliver?
...
+          GitReflow.say "Merging pull request ##{self.number}: '#{self.title}', from '#{self.feature_branch_name}' into '#{self.base_branch_name}'", :notice
 
-          unless base_branch_name.index(':').nil?
-            base = base_branch_name[(base_branch_name.index(':') + 1)..-1]
-          end
-
-          unless feature_branch_name.index(':').nil?
-            feature = feature_branch_name[(feature_branch_name.index(':') + 1)..-1]
-          end
+          self.base_branch_name = self.base_branch_name[/[^:]+$/]
```
Comment (codenamev) [inline lib/git_reflow/git_server/pull_request.rb]: why do we need to check for `:`?

Comment (simonzhu24) [inline lib/git_reflow/git_server/pull_request.rb]: @codenamev Line 180 is the same logic as what I am trying to do previously in line 180 - 186. This is just a bit more elegant Ruby Regex Syntax to match the expression in the string after the ":".

There are 2 cases: 
"master" => "master"
"codenamev:master" => "master"

Regex supports both ^_^

**Comment by simonzhu24** [comment]
The additional tests make sure that each command that was supposed to run executed on the correct `branch_name.` It should be `master` instead of `codenamev:master` since github does not recognize the latter syntax.

File: lib/git_reflow/git_server/pull_request.rb (hunk 1 above)
Comment (codenamev) [inline lib/git_reflow/git_server/pull_request.rb]: Ah, I wasn't aware that Github or Bitbucket formatted the head/base labels like that.  Let's move this into the `GitHub::PullRequest#initialize` and `BitBucket::PullRequest#initialize` methods.  We should be able to trust a consistent format wherever else we need to reference these.

Comment (simonzhu24) [inline lib/git_reflow/git_server/pull_request.rb]: Done

**Comment by codenamev** [comment]
lgtm, thank you :smile:

**Comment by simonzhu24** [comment]
@codenamev np, lets merge this!

**Pull request merged**